CoreMapView/src/au/gov/asd/tac/constellation/views/mapview2/plugins/ExtractCoordsFromGraphPlugin.java

```diff
@@ -0,0 +1,235 @@
+
+        if (graph != null) {
+
+            final GraphElementType[] elementTypes = new GraphElementType[]{GraphElementType.VERTEX, GraphElementType.TRANSACTION};
+
+            mapViewTopComponent.getMapViewPane().getMap().clearQueriedMarkers();
+
+            try {
+                for (final GraphElementType elementType : elementTypes) {
+                    // Ids for all attributes needed from a single vertext of a graph
+                    int lonID = GraphConstants.NOT_FOUND;
+                    int latID = GraphConstants.NOT_FOUND;
+                    int colourID = GraphConstants.NOT_FOUND;
+                    int blazeID = GraphConstants.NOT_FOUND;
+                    int overlayID = GraphConstants.NOT_FOUND;
+                    int labelAttrID = GraphConstants.NOT_FOUND;
+                    int identifierID = GraphConstants.NOT_FOUND;
+
+                    int elementCount;
+
+                    switch (elementType) {
+                        case VERTEX:
+                            // Get IDs
+                            lonID = SpatialConcept.VertexAttribute.LONGITUDE.get(graph);
+                            latID = SpatialConcept.VertexAttribute.LATITUDE.get(graph);
+                            colourID = VisualConcept.VertexAttribute.COLOR.get(graph);
+                            blazeID = VisualConcept.VertexAttribute.BLAZE.get(graph);
+                            overlayID = VisualConcept.VertexAttribute.OVERLAY_COLOR.get(graph);
+                            labelAttrID = VisualConcept.VertexAttribute.LABEL.get(graph);
+                            identifierID = VisualConcept.VertexAttribute.IDENTIFIER.get(graph);
+                            elementCount = graph.getVertexCount();
+                            break;
+                        case TRANSACTION:
+                            lonID = SpatialConcept.VertexAttribute.LONGITUDE.get(graph);
+                            latID = SpatialConcept.VertexAttribute.LATITUDE.get(graph);
+                            elementCount = graph.getTransactionCount();
+                            break;
+                        default:
+                            continue;
+                    }
+
+                    // Loop though every graph element
+                    for (int elementPos = 0; elementPos < elementCount; ++elementPos) {
+                        int elementID = -99;
+
+                        switch (elementType) {
+                            case VERTEX:
+                                elementID = graph.getVertex(elementPos);
+
+                                break;
+                            case TRANSACTION:
+                                elementID = graph.getTransaction(elementPos);
+                                break;
+                            default:
+                                break;
```

Comment:
since the default case won't be hit (you've covered all the possible cases), I would remove the default -99 result and just use a ternary statement instead of a switch. i.e.
`final int elementID = elementType == GraphElementType.VERTEX ? graph.getVertex(elementPos) : graph.getTransaction(elementPos)`